admin-dev/themes/new-theme/js/notifications.js

```diff
@@ -69,20 +69,48 @@ let fillTpl = function (results, eltAppendTo, tpl) {
         return;
       }
 
-      eltAppendTo.children('.notification-elements').append(
-        tpl.replace(/_id_order_/g, parseInt(value.id_order))
-          .replace(/_customer_name_/g, value.customer_name)
-          .replace(/_iso_code_/g, value.iso_code)
-          .replace(/_carrier_/g, (value.carrier !== "" ? ` - ${value.carrier}` : ""))
-          .replace(/_total_paid_/g, value.total_paid)
-          .replace(/_id_customer_/g, parseInt(value.id_customer))
-          .replace(/_company_/g, (value.company !== "" ? ` (${value.company}) ` : ""))
-          .replace(/_date_add_/g, value.date_add)
-          .replace(/_status_/g, value.status)
-          .replace(/order_url/g, `${baseAdminDir}index.php?tab=AdminOrders&token=${token_admin_orders}&vieworder&id_order=${value.id_order}`)
-          .replace(/customer_url/g, `${baseAdminDir}index.php?tab=AdminCustomers&token=${token_admin_customers}&viewcustomer&id_customer=${value.id_customer}`)
-          .replace(/message_url/g, `${baseAdminDir}index.php?tab=AdminCustomerThreads&token=${token_admin_customer_threads}&viewcustomer_thread&id_customer_thread=${value.id_customer_thread}`)
-      );
+      const router = new Router();
+      let tplReplaced = '';
+      let route = '';
+
+      tplReplaced = tpl
+        .replace(/_id_order_/g, parseInt(value.id_order, 10))
+        .replace(/_customer_name_/g, value.customer_name)
+        .replace(/_iso_code_/g, value.iso_code)
+        .replace(/_carrier_/g, (value.carrier !== '' ? ` - ${value.carrier}` : ''))
+        .replace(/_total_paid_/g, value.total_paid)
+        .replace(/_company_/g, (value.company !== '' ? ` (${value.company}) ` : ''))
+        .replace(/_date_add_/g, value.date_add)
+        .replace(/_id_customer_/g, parseInt(value.id_customer, 10))
+        .replace(/_company_/g, (value.company !== '' ? ` (${value.company}) ` : ''))
+        .replace(/_date_add_/g, value.date_add)
+        .replace(/_status_/g, value.status);
+
+      switch (eltAppendTo[0].id) {
+        case 'orders-notifications': {
+          const token = `${window.token_admin_orders}&vieworder&id_order=${value.id_order}`;
```

Comment:
There are two things i don't understand here:
- why is the id_order parameter included in the token const?
- since the page is migrated why not use the `admin_orders_view` route like in customers and messages

Reply:
For your first point, I think it should work, but yes the name of the variable is misleading since the `token` const contains the token but also following query parameters... it should probably be split into 2 variables.

Reply:
It should work indeed, thanks to the legacy_link redirection But it doesn't mean we should do it We have a proper symfony route and a router component in js so we must use them for migrated pages
I think it should also manage the token so it solves all the issues of this code